tools/scylla-sstable.cc

```diff
@@ -2137,11 +2158,12 @@ Validate the specified sstables:
         ("system-schema", bpo::value<sstring>(), "table has to be a system table, name has to be in `keyspace.table` notation")
         ;
 
-    app.add_positional_options({
-        {"sstables", bpo::value<std::vector<sstring>>(), "sstable(s) to process, can also be provided as positional arguments", -1},
-    });
-
     if (found_op) {
+        if (found_op->kind() == operation::kind::read) {
+            app.add_positional_options({
+                {"sstables", bpo::value<std::vector<sstring>>(), "sstable(s) to process, can also be provided as positional arguments", -1},
+            });
+        }
```

Comment:
This runtime construction kind of runs --help. I don't see a way to fix it with boost program_options though, it's too outdated. Python has great support for it via add_subparsers().

Reply:
Options added this way still appear in the `--help` output. In fact that is the entire point in playing these games here: I want only the options relevant to a given command to appear in its `--help` output.

Reply:
You are right that these won't appear in `scylla sstable --help`, but they will in `scylla sstable dump-data --help`, an operation to which the `sstables` option is relevant.

Reply:
Ah, so it's not too bad. In any case we can't do much about it.